### PR TITLE
SmartIP + refactor

### PR DESCRIPTION
**Pull request by arrudagates**
This PR will add the custom multisig pallet and the SmartIP pallet, will also refactor the codebase to remove unnecessary crates and change IPT to IPF

### CODE REVIEW

**Comment by arrudagates** [comment]
TODO:

- [ ] Refactor multisig pallet to remove anything unnecessary for InvArch and rewrite what's necessary
- [ ] Properly set weights for everything on pallet_multisig, pallet_smartip and pallet_ips
- [ ] Remove buy, sell, list functions from pallet_ips
- [ ] Write mocks + tests for pallet_multisig and pallet_smartip, rewrite tests on pallet_ips
- [ ] Implement the WASM to WAT + mint IPF of WAT in the pallet_smartip create function ([Reference](https://github.com/InvArch/InvArch-Pallet-Library/blob/gabriel-smartip%2Brefactor/smartip/src/lib.rs#L117-L118))
- [ ] Implement complementary functions on pallet_smartip (delete, amend, etc)
- [ ] On pallet_ips create function, create asset using the IPS account instead of the call author
- [ ] Implement ownership list in the SmartIP create and IPS create functions, defaulting to caller having 100% ownership if not supplied (necessary for the checkbox above)
- [ ] Add a boolean to the IPS structure to identify if it's a SmartIP or a normal IPS

File: ipf/README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 [![Compatible with Substrate v3.0.0](https://img.shields.io/badge/Substrate-v3.0.0-E6007A)](https://github.com/paritytech/substrate/releases/tag/v3.0.0)
 
-# IPT Pallet: IP Tokens for Substrate
+# IPF Pallet: IP Tokens for Substrate
```
Comment (SHA888) [inline ipf/README.md:3]: Need to change IP Tokens to IP Files

File: ipf/README.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 This is a [Pallet](https://substrate.dev/docs/en/knowledgebase/runtime/pallets) that defines basic functions
 to create and manage [intellectual property (IP)](https://en.wikipedia.org/wiki/Intellectual_property) stored as [non-fungible tokens (NFTs)](https://en.wikipedia.org/wiki/Non-fungible_token). 
 
-# IPTokens : Non-fungible components that define an idea.
+# IPFokens : Non-fungible components that define an idea.
```
Comment (SHA888) [inline ipf/README.md:8]: Same as above (IPFokens to IPFiles)

File: ipf/README.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
 The following **components** are defined:
-* `IPToken` + Metadata
+* `IPFoken` + Metadata
```
Comment (SHA888) [inline ipf/README.md:11]: IPFiles

File: ipf/README.md [hunk 4]
```diff
@@ -18,9 +18,9 @@ The following **functions** are possible:
 
 # IP Token
 
-An IP Token (IPT) is a part of a set, and can be thought of as a component of an idea. Either by itself or in combination with other IP Tokens, it serves to strengethen the foundation for an innovation. IP Tokens represent a unique digital asset.
+An IP Token (IPF) is a part of a set, and can be thought of as a component of an idea. Either by itself or in combination with other IP Tokens, it serves to strengethen the foundation for an innovation. IP Tokens represent a unique digital asset.
```
Comment (SHA888) [inline ipf/README.md:21]: Same as above

**Review by SHA888 — CHANGES_REQUESTED** [review]
I guess it just needs a small tweak before the SmartIP is ready to merge, right?

**Pull request merged**